.circleci/config.yml

```diff
@@ -210,6 +219,72 @@ jobs:
                   [ -n "<< pipeline.parameters.PACKAGE_INSTALL_COMMAND >>" ] && eval "<< pipeline.parameters.PACKAGE_INSTALL_COMMAND >>"
                   eval "<< pipeline.parameters.BUILD_COMMAND >>"
 
+      - when:
+          condition:
+            equal: [true, "<< pipeline.parameters.IS_NEXT_JS >>"]
+          steps:
+            - run:
+                name: Build Next.js Project
+                working_directory: ~/project/repository
+                command: |
+                  set -e
+
+                  # Function to log messages
+                  log() {
+                    echo "$(date +'%Y-%m-%d %H:%M:%S') - $1"
+                  }
+
+                  # Load environment variables
+                  log "Loading environment variables..."
+                  [ -f ~/project/.env ] && source ~/project/.env
+                  [ -f ~/project/.env.ef_circleci ] && source ~/project/.env.ef_circleci
+
+                  # Change to the specified build directory if provided
+                  if [ -n "<< pipeline.parameters.BUILD_DIR >>" ]; then
+                    log "Changing to build directory: << pipeline.parameters.BUILD_DIR >>"
+                    cd "<< pipeline.parameters.BUILD_DIR >>"
+                  fi
+
+                  # Install dependencies
+                  if [ -n "<< pipeline.parameters.PACKAGE_INSTALL_COMMAND >>" ]; then
+                    log "Installing dependencies..."
+                    eval "NEXT_PRIVATE_STANDALONE=true << pipeline.parameters.PACKAGE_INSTALL_COMMAND >>"
+                  else
+                    log "Error: PACKAGE_INSTALL_COMMAND is not set. Exiting."
+                    exit 1
+                  fi
+
+                  # Build the project
+                  if [ -n "<< pipeline.parameters.BUILD_COMMAND >>" ]; then
+                    log "Building the project..."
+                    eval "NEXT_PRIVATE_STANDALONE=true << pipeline.parameters.BUILD_COMMAND >>"
+                  else
+                    log "Error: BUILD_COMMAND is not set. Exiting."
+                    exit 1
+                  fi
+
+                  # Create output directory
+                  log "Creating output directories..."
+                  mkdir -p output output/public output/.next output/node_modules
+
+                  # Move necessary files from standalone to output
+                  log "Moving files to output directory..."
+                  mv .next/standalone/.next output/
+                  mv .next/standalone/node_modules output/
```

Comment:
Do we need node_modules in the output?

Reply:
This `node_modules` is inside the `.next/standalone` folder. It is a compressed version of `node_modules`; it is not the same as the one generated with `npm install`

Reply:
oh i see. why not just copy everything inside .next/standalone to output?

Reply:
Cause we also need to move `next/static/` and `next/public`, it should be like this
<img width="132" alt="image" src="https://github.com/user-attachments/assets/8094e79c-e0fc-49f1-9b50-65e38ccb172f" />

Reply:
I understand, I'm saying copy .next/standalone into output then copy public and .next/static also. That way if output standalone adds a new folder in .next /standalone in the future it would copy that by default

Reply:
ohh I got, going to try again but I think I was having some issue doing that cause .next hidden folder, but going to try again